lib/ansible/modules/unarchive.py

```diff
@@ -937,7 +950,10 @@ def pick_handler(src, dest, file_args, module):
         if can_handle:
             return obj
         reasons.add(reason)
-    reason_msg = ' '.join(reasons)
+    joiner = ' '
+    if module._verbosity >= 3:
+        joiner = '\n'
+    reason_msg = joiner.join(reasons)
```

Comment:
Maybe a better approach would be to have the `can_handle_archive()` method return errors explicitly then, pass those in to `fail_json()` as `stderr` or `module_stderr` rather than stuffing everything inside `msg`.